[Accordion] Allow to disable gutter/spacing

TimonPllkrn opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

resolves #24492 

I've added the `disableGutters` prop to the `AccordionSummary` and `AccordionDetails` Components.

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui-org/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

oliviertassinari commented on packages/material-ui/src/AccordionDetails/AccordionDetails.d.ts:
```diff
@@ -12,7 +12,13 @@ export interface AccordionDetailsProps extends StandardProps<React.HTMLAttribute
   classes?: {
     /** Styles applied to the root element. */
     root?: string;
+    gutters?: string;
```
We miss a description
```suggestion
    /** Styles applied to the root element unless `disableGutters={true}`. */
    gutters?: string;
```

oliviertassinari commented on packages/material-ui/src/AccordionDetails/AccordionDetails.js:
```diff
@@ -6,14 +6,24 @@ import withStyles from '../styles/withStyles';
-    padding: theme.spacing(1, 2, 2),
+    padding: theme.spacing(1, 0, 2),
+  },
+  gutters: {
```
```suggestion
  /** Styles applied to the root element unless `disableGutters={true}`. */
  gutters: {
```

oliviertassinari reviewed (changes requested):
I don't think that we are on the same page. The objective was to disable the very specific behavior of Material Design that adds a margin between to accordion items and that changes the margin inside one item:

e.g. 

<img width="356" alt="Capture d’écran 2021-01-21 à 20 04 35" src="https://user-images.githubusercontent.com/3165635/105399449-ede8cd80-5c23-11eb-8cde-2e55994540e4.png">

TimonPllkrn commented:
Ok, I was confused because disableGutters in the other components means remove padding left and right.
Implemented it like this now:

`<Accordion disableGutters>`
![accordion_disableGutters](https://user-images.githubusercontent.com/64217916/105475251-75bfed80-5c9f-11eb-960d-ce101333a498.gif)

`<Accordion>`
![accordion](https://user-images.githubusercontent.com/64217916/105475278-7ce6fb80-5c9f-11eb-8dcb-3e28f90a29c6.gif)

oliviertassinari commented:
@TimonPllkrn Yes, I think that this is the behavior that makes the most sense :). If you have ideas for improving the name of the prop, I would love to hear them. I also agree that it's confusing.

The padding of the component should be easy enough to customize, I don't think that we need new dedicated props for them.

Mind that the component is being migrated to emotion, I would encourage us to hold on until these pull requests are merged.

<img width="685" alt="Capture d’écran 2021-01-22 à 12 09 33" src="https://user-images.githubusercontent.com/3165635/105483938-e15b8800-5caa-11eb-826c-1af781ae2557.png">

mnajdova commented:
@TimonPllkrn @oliviertassinari all PRs for migrating the Accordion components are merged. This work can continue :)

TimonPllkrn commented:
I couldn't think of a better name either.

I have now adapted the changes to the migration and added `disableGutters` to the demo of `Customized accordion`.

Also added the classes gutters and contentGutters to customize the style.

oliviertassinari reviewed (approved):
I have used the opportunity to polish the customization demo

<img width="792" alt="Capture d’écran 2021-01-24 à 22 12 48" src="https://user-images.githubusercontent.com/3165635/105643646-57b9df00-5e91-11eb-9d38-4b2e6295feea.png">

mnajdova reviewed (approved):
👍
